internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/DiagnosticsReporter.scala

```diff
@@ -104,7 +104,13 @@ object DiagnosticsReporter {
         }
 
       val source: Option[JavaFileObject] = Option(d.getSource)
-      val sourcePath: Option[String] = source map (obj => IO.toFile(obj.toUri).getAbsolutePath)
+      val sourcePath: Option[String] = source.map { obj =>
+        val uri = obj.toUri
+        if (uri.isAbsolute && uri.getScheme == "file")
+          IO.toFile(uri).getAbsolutePath
+        else
+          uri.toString
```

Comment:
The error case reported on the original issue was `jar:file:///C:/Users/user/.ivy2/cache/com.diffplug.durian/durian/jars/durian-3.4.0.jar!/com/diffplug/common/base/Errors.class`. In this case, I think it would be good to extract the `file` portion out of `jar:` and return the absolute path for `durian-3.4.0.jar`. There's some code that splits the jar part and the in-jar path in https://github.com/sbt/zinc/blob/develop/internal/zinc-classfile/src/main/scala/sbt/internal/inc/JarUtils.scala.

Reply:
JarUtils didn't seem to have the splitting logic needed to handle this so I went with using `IO.urlAsFile` in the case where the URI is absolute.

https://github.com/sbt/io/blob/d3297b56297a524baf5982661170edfdb8a6823e/io/src/main/scala/sbt/io/IO.scala#L237-L245